src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresViewIT.java

```diff
@@ -49,75 +56,120 @@
 
   @After
   public void after() throws SQLException {
-    stopTask();
     try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
       try (Statement s = c.createStatement()) {
         s.execute("DROP VIEW " + topic);
         s.execute("DROP TABLE " + tableName);
       }
     }
     log.info("Dropped table");
+    stopConnect();
   }
 
+  /**
+   * Verifies that when sending records with more fields than the view has, these errant records
+   * are sent to the error reporter. The test also intersperses correct schema records to verify
+   * that only the errant records are being sent to the error reporter.
+   */
   @Test
-  public void testRecordSchemaMoreFieldsThanViewFails() throws SQLException {
-    createTestTableAndView("firstName");
-    startTask();
-    final Schema schema = SchemaBuilder.struct().name("com.example.Person")
+  public void testRecordSchemaMoreFieldsThanViewSendsToErrorReporter() throws Exception {
+    props.put(ERRORS_TOLERANCE_CONFIG, ToleranceType.ALL.value());
+    props.put(DLQ_TOPIC_NAME_CONFIG, DLQ_TOPIC_NAME);
+    props.put(DLQ_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+
+    createTestTableAndView("firstName, lastName");
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
+    final Schema correctSchema = SchemaBuilder.struct().name("com.example.Person")
+        .field("firstname", Schema.STRING_SCHEMA)
+        .field("lastname", Schema.STRING_SCHEMA)
+        .build();
+    final Struct correctStruct = new Struct(correctSchema)
+        .put("firstname", "Christina")
+        .put("lastname", "Brams");
+
+    final Schema errorSchema = SchemaBuilder.struct().name("com.example.Person")
         .field("firstname", Schema.STRING_SCHEMA)
         .field("lastname", Schema.STRING_SCHEMA)
         .field("age", Schema.INT32_SCHEMA)
         .build();
-    final Struct struct = new Struct(schema)
+    final Struct errorStruct = new Struct(errorSchema)
         .put("firstname", "Christina")
         .put("lastname", "Brams")
         .put("age", 20);
-    try {
-      task.put(Collections.singleton(new SinkRecord(topic, 1, null, null, schema, struct, 1)));
-      fail();
-    } catch (ConnectException e) {
-      assertTrue(e.getMessage().contains("View \"" + topic + "\" is missing fields"));
+
+    for (int i = 0; i < 6; i++) {
+      if (i % 2 == 0) {
+        produceRecord(correctSchema, correctStruct);
+      } else {
+        produceRecord(errorSchema, errorStruct);
+      }
     }
+
+    waitForCommittedRecords("jdbc-sink-connector", Collections.singleton(topic), 6, 1,
+        TimeUnit.MINUTES.toMillis(4));
+
+    KafkaConsumer<byte[], byte[]> consumer =
+        connect.kafka().createConsumerAndSubscribeTo(Collections.emptyMap(), DLQ_TOPIC_NAME);
+
+    ConsumerRecords<byte[], byte[]> records =
+        consumer.poll(Duration.ofMillis(CONSUME_MAX_DURATION_MS));
+
+    assertEquals(3, records.count());
```

Comment:
Same question here w/r/t manually constructing the consumer.

Reply:
Fixed this.